gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/api/model/ApiMetadata.java

```diff
@@ -18,16 +18,15 @@
 import io.gravitee.rest.api.model.MetadataFormat;
 import lombok.AllArgsConstructor;
 import lombok.Builder;
-import lombok.Data;
 import lombok.NoArgsConstructor;
 import lombok.With;
 
 @Builder
-@Data
 @AllArgsConstructor
 @NoArgsConstructor
 public class ApiMetadata {
 
+    String apiId;
```

Comment:
I'm not sure if there's an added value to having `apiId` in this object. For now, we do not return the apiId in mapi-v2 in the response.

From what I can see in the old services, the metadata entities are the same. This class could then be renamed `Metadata` instead of `ApiMetadata`.

Reply:
It is a deliberate choice that I created two models:
`ApiMetadata` is a view of `Metadata` collections. As you can see there is a `defaultValue` attribute that is not present in `Metadata` because it is calculated by https://github.com/gravitee-io/gravitee-api-management/blob/refactor-extract-create-api-usecase-run-e2e/gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/query_service/api/ApiMetadataQueryServiceImpl.java#L45

`Metadata` can be seen as a Create/Update model and `ApiMetadata` as a view model.

Reply:
Ok I can see that reasoning. The `apiId` for me is still superfluous, but that makes sense to keep Metadata + ApiMetadata separate.

Reply:
I've added the attribute thinking I could rely on `ApiMetadata` when creating the default metadata when creating the API, but I kept it because simplifies the usage of the InMemory implementation (see https://github.com/gravitee-io/gravitee-api-management/pull/6659/commits/c0ed676bc31fbbd429add90fccfebcd953987656)